projects/wave-core-new/src/lib/layers/layer.model.ts

```diff
@@ -30,72 +35,112 @@ export abstract class Layer implements ToDict<LayerDict> {
     }
 
     protected constructor(config: {
```

Comment:
could we use interfaces for the config? in case we need it somewhere else?

Reply:
We could, but I personally find it easier to use if its right in the constructor (like regular parameters).